elements/lisk-transaction-pool/src/queue_checkers.ts

```diff
@@ -10,8 +11,22 @@ export type TransactionFilterableKeys =
 export const checkTransactionPropertyForValues = (
 	values: ReadonlyArray<string | number>,
 	propertyName: TransactionFilterableKeys,
-): ((transaction: Transaction) => boolean) => (transaction: Transaction) =>
-	values.includes(transaction[propertyName]);
+): ((transaction: Transaction) => boolean) => (transaction: Transaction) => {
+	if (propertyName === 'recipientId') {
+		return transaction.asset.recipientId &&
+			typeof transaction.asset.recipientId === 'string'
+			? values.includes(transaction.asset.recipientId)
+			: false;
+	}
+
+	if (propertyName === 'senderId') {
+		return values.includes(
+			getAddressFromPublicKey(transaction.senderPublicKey),
+		);
+	}
+
```

Comment:
do we also need a check for `asset.amount`?

Reply:
I think it only searches by recipient/sender never per amount but @shuse2 can you confirm bitte?